ompi/mca/coll/hcoll/coll_hcoll_module.c

```diff
@@ -138,7 +138,7 @@ static void mca_coll_hcoll_module_destruct(mca_coll_hcoll_module_t *hcoll_module
         OBJ_RELEASE(hcoll_module->previous_reduce_scatter_module);
         OBJ_RELEASE(hcoll_module->previous_reduce_module);
         */
-#if HCOLL_API < HCOLL_VERSION(3,7)
+#ifndef HAVE_HCOLL_CONTEXT_FREE
```

Comment:
please use #ifdef style instead #ifndef for better readability
Also, need incude config.h for HAVE_xxx symbol